### ISSUE TITLE
Prompt padding inconsistency in updating rolling state

### ISSUE THREAD

**Issue by rhinosaur0** [TIGER-AI-Lab/verl-tool#79]
This happens when I'm using async rollout
When updating rolling state, the original_left_side ids seems to be padded to the max prompt length, while the rolling state ids don't. I'm not sure if this only happened to me because I modified some logic, but all the variables I was using seemed to be the same as the version in the repo.

**Comment by rhinosaur0** [comment]
My max prompt length is set at 4500.
```
# before generate_sequences
--- rollings.batch['input_ids'] before first generation (shape: torch.Size([492])) ---
<|im_start|>system...

# in update_rolling_state
--- Final concatenated context (shape: torch.Size([1, 1402])) ---
<|im_start|>system...

# in update_rolling_state
--- Original left (shape: torch.Size([1, 4500])) --- 
<|endoftext|><|endoftext|><|endoftext|><|endoftext|>...
```


I think it's this part below that is causing it.
`manager.py`
```python       
rollings.batch = self.tensor_fn.cut_to_effective_len(
                rollings.batch,
                keys=['input_ids', 'attention_mask', 'position_ids']
            ) # TODO: delete
```

It's removing the padding of my prompts. During the update_rolling_state below, when I have rolling_with_prompts set to True, the padded original prompt (4500) exceeds my new prompt-response-observation context (1402), causing the new context to just be the last 1402 chars of the prompt, which is just a bunch of pad tokens at the front (since it's left padded) and the original prompt. 

```python
        effective_lens = new_attention_mask.sum(dim=1)
        effective_len = effective_lens.max()
        min_effective_len = effective_lens.min().item()
        # max_len = min(self.config.max_prompt_length, effective_len)
        max_len = min(self.config.max_prompt_length+self.config.max_response_length, effective_len)
        available_context_budget = max(0, self.config.max_prompt_length+self.config.max_response_length - min_effective_len)
        assert isinstance(available_context_budget, int), f"available_context_budget should be int, but got {type(available_context_budget)}"
        if getattr(self.config, "rolling_with_prompt", False):
            # if rolling_with_prompt is True, then we need to keep the system prompt
            if isinstance(left_side, dict):
                left_ids = left_side["input_ids"]
            else:
                left_ids = left_side.batch["input_ids"]

            left_len = left_ids.size(1)
            logger.warning(f"left len in rolling {left_len}")

            if left_len >= max_len:
                final_input_ids = left_ids[:, -max_len:]
            else:
                right_budget = max_len - left_len
                right_ids_full = new_input_ids[:, left_len:]
                right_ids = right_ids_full[:, -right_budget:] if right_budget < right_ids_full.size(1) else right_ids_full
                final_input_ids = torch.cat([left_ids, right_ids], dim=1)
```
